src/main/java/org/testng/internal/DynamicGraph.java

```diff
@@ -50,64 +49,103 @@ public void addNode(T node) {
   /**
    * Add an edge between two nodes.
    */
-  public void addEdge(T from, T... tos) {
-    addEdge(from, Arrays.asList(tos));
+  public void addEdge(int weight, T from, T... tos) {
+    addEdge(weight, from, Arrays.asList(tos));
   }
 
   /**
    * Add an edge between two nodes.
    */
-  public void addEdge(T from, Iterable<T> tos) {
+  public void addEdge(int weight, T from, Iterable<T> tos) {
```

Comment:
I think it would be more consistent to have `(T from, Iterable<T> tos, int weight)` since that's the ordering that `Edge` expects.

Also, you could add an overload for `addEdge()` that sets `weight` to 0 (not sure it's useful yet, not done reviewing everything).